src/log/Log.cc

```diff
@@ -312,13 +313,13 @@ void Log::_flush(EntryVector& t, bool crash)
         syslog(LOG_USER|LOG_INFO, "%s", pos);
       }
 
-      if (do_stderr) {
-        std::cerr << m_log_stderr_prefix << std::string_view(pos, used) << std::endl;
-      }
-
       /* now add newline */
       pos[used++] = '\n';
 
+      if (do_stderr) {
+        std::cerr << fmt::format("{}{}", m_log_stderr_prefix, std::string_view(pos, used));
+      }
```

Comment:
please use https://fmt.dev/latest/api.html#_CPPv4IDpEN3fmt5printEvRNSt7ostreamE13format_stringIDp1TEDpRR1T instead, like

```suggestion
      if (do_stderr) {
        fmt::format(std::cerr, "{}{}", m_log_stderr_prefix, std::string_view(pos, used));
      }
```

Reply:
just saw https://github.com/ceph/ceph/pull/44356#issuecomment-999403459 , closing. sorry for the noise.